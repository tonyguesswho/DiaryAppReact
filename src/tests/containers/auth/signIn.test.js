import React from 'react';
import { mount, shallow } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import SigninComponent, { SignIn } from '../../../containers/auth/SignIn';
import Root from '../../../root';

let wrapped;
let wrapper;

const props = {
  clearError: jest.fn()
}

beforeEach(() => {
  wrapped = mount(
    <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
      <Root>
        <SigninComponent
          clearError={props.clearError}
        />
      </Root>
    </MemoryRouter>
  );

  wrapper = shallow(
    <SignIn
      clearError={props.clearError}
    />
  );
});

afterEach(() => wrapped.unmount());

describe('Signup UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });

    it('has 3 input fields and a button', () => {
      expect(wrapped.find('input').length).toEqual(3);
    });

    it('should simulate signin process', () => {
      const Signin = wrapper.find('form');

      const mockedEvent = {
        preventDefault: jest.fn()
      }

      Signin.simulate('submit', mockedEvent)
    });

    it('should simulate onChange validation process', () => {
      const Signin = wrapper.find('#password');

      const mockedEvent = {
        preventDefault: jest.fn(),
        target: {
          name: 'name',
          value: 'value'
        }
      }

      Signin.simulate('change', mockedEvent)
    });
  });
});
