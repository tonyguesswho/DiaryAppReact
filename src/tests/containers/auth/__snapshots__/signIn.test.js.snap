// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup UI render features container should render as expected 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <_default>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(withRouter(SignIn))
          clearError={
            [MockFunction] {
              "calls": Array [
                Array [],
              ],
              "results": Array [
                Object {
                  "isThrow": false,
                  "value": undefined,
                },
              ],
            }
          }
        >
          <withRouter(SignIn)
            clearError={[Function]}
            errorMessage=""
            signin={[Function]}
          >
            <Route>
              <SignIn
                clearError={[Function]}
                errorMessage=""
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/",
                        "search": "",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
                signin={[Function]}
              >
                <div
                  className="bg body"
                >
                  <header
                    className="header2"
                  >
                    <nav>
                      <ul
                        className="nav nav2"
                      >
                        <li>
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              Home
                            </a>
                          </Link>
                        </li>
                        <li>
                          <Link
                            replace={false}
                            to="/signup"
                          >
                            <a
                              href="/signup"
                              onClick={[Function]}
                            >
                              Sign Up
                            </a>
                          </Link>
                        </li>
                      </ul>
                    </nav>
                  </header>
                  <section
                    className="signup"
                    id="messageBox"
                  >
                    <div
                      className="bigboxy"
                    >
                      <div
                        className="box span31 span3-center card"
                      >
                        <h2>
                          Sign In 
                        </h2>
                        <form
                          id="login"
                          onSubmit={[Function]}
                        >
                          <div
                            className="fullspan"
                          >
                            <label
                              htmlFor="email"
                            >
                              Email:
                            </label>
                          </div>
                          <div
                            className="fullspan"
                          >
                            <input
                              id="email"
                              name="email"
                              onChange={[Function]}
                              placeholder="email"
                              required={true}
                              type="email"
                            />
                          </div>
                          <div
                            className="fullspan"
                          >
                            <label
                              htmlFor="password"
                            >
                              Password:
                            </label>
                          </div>
                          <div
                            className="fullspan"
                          >
                            <input
                              id="password"
                              name="password"
                              onChange={[Function]}
                              placeholder="password"
                              required={true}
                              type="password"
                            />
                          </div>
                          <p
                            className="signin-prompt"
                          >
                            Don't have an account
                            <Link
                              replace={false}
                              to="/signup"
                            >
                              <a
                                href="/signup"
                                onClick={[Function]}
                              >
                                Sign Up
                              </a>
                            </Link>
                          </p>
                          <input
                            id="signin"
                            type="submit"
                            value="Sign In"
                          />
                        </form>
                      </div>
                    </div>
                  </section>
                </div>
              </SignIn>
            </Route>
          </withRouter(SignIn)>
        </Connect(withRouter(SignIn))>
      </Provider>
    </_default>
  </Router>
</MemoryRouter>
`;
